packages/@aws-cdk/aws-apigateway/lib/domain-name.ts

```diff
@@ -102,17 +110,25 @@ export class DomainName extends Resource implements IDomainName {
 
     const endpointType = props.endpointType || EndpointType.REGIONAL;
     const edge = endpointType === EndpointType.EDGE;
-
+    const getMTLSConfiguration = (mtlsConfig?: MTLSAttributes): CfnDomainName.MutualTlsAuthenticationProperty | undefined => {
```

Comment:
Pushed a commit where I tried this just to verify I've done this in the way you expected.
I moved getMTLSConfiguration into a function instead of being in the constructor (note, I only put the function here to make this diff clean, I was going to put it elsewhere).
```diff
diff --git a/packages/@aws-cdk/aws-apigateway/lib/domain-name.ts b/packages/@aws-cdk/aws-apigateway/lib/domain-name.ts
index a59b71892..b490c1c80 100644
--- a/packages/@aws-cdk/aws-apigateway/lib/domain-name.ts
+++ b/packages/@aws-cdk/aws-apigateway/lib/domain-name.ts
@@ -86,6 +86,14 @@ export interface IDomainName extends IResource {
 
 }
 
+function getMTLSConfiguration(mtlsConfig?: MTLSConfig): CfnDomainName.MutualTlsAuthenticationProperty | undefined {
+  if (!mtlsConfig) return undefined;
+  return {
+    truststoreUri: mtlsConfig.bucket.s3UrlForObject(mtlsConfig.key),
+    truststoreVersion: mtlsConfig.version,
+  };
+};
+
 export class DomainName extends Resource implements IDomainName {
 
   /**
@@ -115,7 +123,7 @@ export class DomainName extends Resource implements IDomainName {
       throw new Error('domainName does not support uppercase letters. ' +
         `got: '${props.domainName}'`);
     }
-    const mtlsConfig = this.getMTLSConfiguration(props.mtls);
+    const mtlsConfig = getMTLSConfiguration(props.mtls);
     const resource = new CfnDomainName(this, 'Resource', {
       domainName: props.domainName,
       certificateArn: edge ? props.certificate.certificateArn : undefined,
@@ -153,14 +161,6 @@ export class DomainName extends Resource implements IDomainName {
       restApi: targetApi,
       ...options,
     });
-  }
```
But now I get a jsii build error
```
[2020-10-06T08:57:54.987] [ERROR] jsii/compiler - Error during type model analysis: TypeError: Cannot read property 'getJsDocTags' of undefined
TypeError: Cannot read property 'getJsDocTags' of undefined
    at _hasInternalJsDocTag (/home/dferris/aws-cdk/node_modules/jsii/lib/assembler.js:1617:19)
    at Assembler._isPrivateOrInternal (/home/dferris/aws-cdk/node_modules/jsii/lib/assembler.js:988:37)
    at Assembler._visitClass (/home/dferris/aws-cdk/node_modules/jsii/lib/assembler.js:834:26)
    at async Assembler._visitNode (/home/dferris/aws-cdk/node_modules/jsii/lib/assembler.js:579:24)
    at async Promise.all (index 86)
    at async Assembler.emit (/home/dferris/aws-cdk/node_modules/jsii/lib/assembler.js:102:9)
    at async Compiler._consumeProgram (/home/dferris/aws-cdk/node_modules/jsii/lib/compiler.js:148:30)
    at async /home/dferris/aws-cdk/node_modules/jsii/bin/jsii.js:70:39
```

I've tried it as both a private function on the class as well as standalone, still get the same behaviour.

As is probably clear, this is my first time working with jsii, so I'm not 100% clear about the why this is occurring.

If you have any suggestions on how to format my code to have the same intent but get around this, it would be greatly appreciated.

Reply:
The problem is likely because you have the `get` prefix. Changing this to `mtlsConfiguration()` should fix this. 

Let me know if not and I'll dig into this.

Reply:
I've tried making this diff locally
```diff
diff --git a/packages/@aws-cdk/aws-apigateway/lib/domain-name.ts b/packages/@aws-cdk/aws-apigateway/lib/domain-name.ts
index a59b71892..71371765b 100644
--- a/packages/@aws-cdk/aws-apigateway/lib/domain-name.ts
+++ b/packages/@aws-cdk/aws-apigateway/lib/domain-name.ts
@@ -86,6 +86,7 @@ export interface IDomainName extends IResource {
 
 }
 
+
 export class DomainName extends Resource implements IDomainName {
 
   /**
@@ -107,7 +108,6 @@ export class DomainName extends Resource implements IDomainName {
 
   constructor(scope: Construct, id: string, props: DomainNameProps) {
     super(scope, id);
-
     const endpointType = props.endpointType || EndpointType.REGIONAL;
     const edge = endpointType === EndpointType.EDGE;
 
@@ -115,13 +115,13 @@ export class DomainName extends Resource implements IDomainName {
       throw new Error('domainName does not support uppercase letters. ' +
         `got: '${props.domainName}'`);
     }
-    const mtlsConfig = this.getMTLSConfiguration(props.mtls);
+    // const mtlsConfig = this.mtlsConfiguration(props.mtls);
     const resource = new CfnDomainName(this, 'Resource', {
       domainName: props.domainName,
       certificateArn: edge ? props.certificate.certificateArn : undefined,
       regionalCertificateArn: edge ? undefined : props.certificate.certificateArn,
       endpointConfiguration: { types: [endpointType] },
-      mutualTlsAuthentication: mtlsConfig,
+      // mutualTlsAuthentication: ,
       securityPolicy: props.securityPolicy,
     });
 
@@ -153,15 +153,15 @@ export class DomainName extends Resource implements IDomainName {
       restApi: targetApi,
       ...options,
     });
-  }
-
-  private getMTLSConfiguration(mtlsConfig?: MTLSConfig): CfnDomainName.MutualTlsAuthenticationProperty | undefined {
-    if (!mtlsConfig) return undefined;
-    return {
-      truststoreUri: mtlsConfig.bucket.s3UrlForObject(mtlsConfig.key),
-      truststoreVersion: mtlsConfig.version,
-    };
   };
+
+  // private mtlsConfiguration(mtlsConfig?: MTLSConfig): CfnDomainName.MutualTlsAuthenticationProperty | undefined {
+  //   if (!mtlsConfig) return { };
+  //   return {
+  //     truststoreUri: mtlsConfig.bucket.s3UrlForObject(mtlsConfig.key),
+  //     truststoreVersion: mtlsConfig.version,
+  //   };
+  // };
 }
```
Still getting the same error.
If you could have a look, that would be fantastic.
I would have thought me commenting out the code would have ruled out if it was my code directly.

Reply:
To clarify the use of `return {}`
I had tried `return undefined` and `return mtlsConfig` and was just trying to try _things_ to try and narrow down the scope of what was causing the break.
Running `yarn build` from inside of the aws-apigateway package threw the error every time, couldn't figure out why it was breaking

Reply:
Sure. I'll take a look.

Reply:
Lol, it seems the problem is the trailing semicolon at the end of the private method definition. JSII seems to not like that. Remove that and you're problem will be fixed 😊

We're tracking this here - https://github.com/aws/jsii/issues/2098